docs/reference/aggregations/bucket/datehistogram-aggregation.asciidoc

```diff
@@ -397,3 +397,55 @@ By default the returned buckets are sorted by their `key` ascending, though the
 the `order` setting. Supports the same `order` functionality as the <<search-aggregations-bucket-terms-aggregation-order,`Terms Aggregation`>>.
 
 deprecated[6.0.0, Use `_key` instead of `_time` to order buckets by their dates/keys]
+
+=== Use of a script to aggregate by day of the week
+
+There are some cases where date histogram can't help us, like for example, when we need
+to aggregate the results by day of the week.
+In this case to overcame the problem, we can use a script that returns the day of the week:
+
+
+[source,js]
+--------------------------------------------------
+POST /sales/_search?size=0
+{
+    "aggs": {
+        "dayOfWeek": {
+            "terms": {
+                "script": {
+                    "inline": "doc['date'].date.dayOfWeek",
+                    "lang": "painless"
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+
```

Comment:
This should be marked as `// CONSOLE` so we automatically test it as part of the build. It should probably also have `// TEST[setup:sales]` below that so that is has some data to chew on.